File: TinyScript/java/src/main/java/common/AlphabetHelper.java
package common;

import java.util.regex.Pattern;

public class AlphabetHelper {

    static Pattern ptnLetter = Pattern.compile("^[a-zA-Z]$");
    static Pattern ptnNumber = Pattern.compile("^[0-9]$");
    static Pattern ptnLiteral = Pattern.compile("^[_a-zA-Z0-9]$");
    static Pattern ptnOperator = Pattern.compile("^[*+\\-<>=!&|^%/,]$");

    public static boolean isLetter(char c) {
        return ptnLetter.matcher(c + "").matches();
    }

    public static boolean isNumber(char c) {
        return ptnNumber.matcher(c + "").matches();
    }

    public static boolean isLiteral(char c) {
        return ptnLiteral.matcher(c + "").matches();
    }

    public static boolean isOperator(char c) {
        return ptnOperator.matcher(c + "").matches();
    }

}

File: TinyScript/unstable/python/lexer/AlphabetHelper.py
# -*- coding: UTF-8 -*-
'''=================================================
@Project -> File   ：TinyScript -> AlphabetHelper
@IDE    ：PyCharm
@Author ：QiangZiBro
@Date   ：2020/4/18 12:55 下午
@Desc   ： judge the type of a character
=================================================='''
import re


class AlphabetHelper:
    ptnLetter = "^[a-zA-Z]$"
    ptnNumber = "^[0-9]$"
    ptnLiteral = "^[_a-zA-Z0-9]$"  # English text
    ptnOperator = r"^[+-\\*/><=!&|^%]$"

    def isLetter(self, c):
        return re.match(self.ptnLetter, c) is not None

    def isNumber(self, c):
        return re.match(self.ptnNumber, c) is not None

    def isLiteral(self, c):
        return re.match(self.ptnLiteral, c) is not None

    def isOperator(self, c):
        return re.match(self.ptnOperator, c) is not None
